packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -430,28 +449,37 @@ class MouseTracker extends ChangeNotifier {
   // The `lastHoverPosition` can be null, which means the last event is not a
   // hover. Other arguments must not be null.
   static void _dispatchDeviceCallbacks({
-    @required LinkedHashSet<MouseTrackerAnnotation> lastAnnotations,
-    @required LinkedHashSet<MouseTrackerAnnotation> nextAnnotations,
+    @required List<MouseTrackerAnnotation> lastAnnotations,
+    @required List<MouseTrackerAnnotation> nextAnnotations,
     @required Offset lastHoverPosition,
     @required PointerEvent unhandledEvent,
-    @required Set<MouseTrackerAnnotation> trackedAnnotations,
+    @required HashSet<Key> trackedAnnotationKeys,
   }) {
     assert(lastAnnotations != null);
     assert(nextAnnotations != null);
     // lastHoverPosition can be null
     assert(unhandledEvent != null);
-    assert(trackedAnnotations != null);
+    assert(trackedAnnotationKeys != null);
     // Order is important for mouse event callbacks. The `findAnnotations`
```

Comment:
How do we keep the order now? Below the code is iterating over non-order preserving iterables, as far as I can tell.

Reply:
The `lastAnnotations` and `nextAnnotations` are still `List`s, which is where order is kept. The `lastKeys` and `nextKeys` below are a `HashSet` because they are only used to check if an annotation is within the last frame / this frame.

There are tests that ensure the order is kept, such as the several tests that start at https://github.com/flutter/flutter/blob/7f4a05dac5ae93bacb1a353f7a194c4083eb4d06/packages/flutter/test/gestures/mouse_tracking_test.dart#L782